lib/heex_formatter/algebra.ex

```diff
@@ -218,16 +211,21 @@ defmodule HeexFormatter.Algebra do
   end
 
   # Handle Text within other tags.
-  defp to_algebra({:text, text, _meta} = node, _context) when is_binary(text) do
-    if newline?(node) do
-      {:newline, empty()}
-    else
-      {:inline,
-       text
-       |> String.split(["\r\n", "\n"])
-       |> Enum.map(&String.trim/1)
-       |> Enum.drop_while(&(&1 == ""))
-       |> text_to_algebra(0, [])}
+  defp to_algebra({:text, text, _meta}, _context) when is_binary(text) do
+    cond do
+      only_spaces?(text) ->
```

Comment:
This handle this case here:

```
<%= @user.first_name %> <%= @user.last_name %>
```

before this code, it would break line between them.

Reply:
Instead of both `only_spaces?` and `String.trim_leading`, you could have a function called: `classify_leading(string)`.

It returns three possible atoms: `:spaces`, `:newline`, :other`. Then you can transform this `cond` into a `case` and traverse the text only once. :)